satpy/readers/clavrx.py

```diff
@@ -450,29 +509,34 @@ def available_datasets(self, configured_datasets=None):
                 # we don't know any more information than the previous
                 # file handler so let's yield early
                 yield is_avail, ds_info
-                continue
-            if self.file_type_matches(ds_info["file_type"]):
+
+            matches = self.file_type_matches(ds_info["file_type"])
+            if matches and ds_info.get("resolution") != self.resolution:
+                # reader knows something about this dataset (file type matches)
+                # add any information that this reader can add.
+                new_info = ds_info.copy()
+                new_info["resolution"] = self.resolution
                 handled_vars.add(ds_info["name"])
-            yield self.file_type_matches(ds_info["file_type"]), ds_info
-        yield from self._available_new_datasets(handled_vars)
+                yield True, new_info
```

Comment:
Should the resolution check here be checking for `is None`? What is the case where we, the file handler, are configured to handle this variable but the resolution is defined as something other than what we contain (currently resolution != self.resolution)?

Reply:
This part has been tricky for me to get correct.  The structure is essentially following [Example 1](https://satpy.readthedocs.io/en/stable/api/satpy.readers.file_handlers.html#satpy.readers.file_handlers.BaseFileHandler.available_datasets).  Are you suggesting:

```python
matches = self.file_type_matches(ds_info["file_type"])
            if matches and ds_info.get("resolution") != self.resolution:
                # reader knows something about this dataset (file type matches)
                # add any information that this reader can add.
                new_info = ds_info.copy()
                if self.resolution is not None:
                    new_info["resolution"] = self.resolution
                handled_vars.add(ds_info["name"])
                yield True, new_info
        yield from self._available_file_datasets(handled_vars)
```
It has become a little complex trying to get the aliases to work within the available datasets, but that has nothing to do with the question of resolution, it just adds layers of yield statements.

Reply:
The git diff is a little complicated now, but I kind of see the reasoning behind the way I made "Example 1". I think the idea is something like MODIS L1b files where each resolution is in a different file (250m, 500m, 1km). So in that case it is very likely that we could be looking at a `ds_info` that has a resolution different than our own and I suppose the file matches is just a "shortcut" condition to say "yeah, we definitely don't know how to handle this dataset".

I guess what I'm saying is leave it the way you have it if it works for you.